pupil-spa/src/app/feedback/feedback.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { StorageService } from '../services/storage/storage.service';
+
+@Component({
+  selector: 'app-feedback',
+  templateUrl: './feedback.component.html',
+  styleUrls: ['./feedback.component.scss']
+})
+export class FeedbackComponent implements OnInit {
+
+  public inputTypes: object;
+  public satisfactionRatings: object;
+  public selectedInputType;
+  public selectedSatisfactionRating;
+  public errorExists: boolean;
+  public errorInputType: boolean;
+  public errorSatisfactionRating: boolean;
+  public errorCommentExists: boolean;
+  public enableSubmit: boolean;
+
+  private pupilData: object;
+  private feedbackExists: boolean;
+  private feedbackData: object;
+  private submitted: boolean;
+
+  constructor(private router: Router, private storageService: StorageService) {}
+
+  ngOnInit() {
+    if (!this.validate()) {
```

Comment:
I find `validate()` confusing in this context (form validation / submission) as really this is checking whether feedback has already been given.  How about calling it `componentValidate()` instead?

Reply:
OK, I've renamed it as suggested to ```componentValidate```